src/Symfony/Component/DependencyInjection/Tests/Compiler/FactoryReturnTypePassTest.php

```diff
@@ -100,4 +102,22 @@ public function testCircularReference()
         $this->assertNull($factory->getClass());
         $this->assertNull($factory2->getClass());
     }
+
+    public function testCompile()
+    {
+        $container = new ContainerBuilder();
+
+        $factory = $container->register('factory');
+        $factory->setFactory(array(FactoryDummy::class, 'createFactory'));
+
+        try {
+            $container->compile();
+        } catch (\RuntimeException $e) {
+            if (method_exists(\ReflectionMethod::class, 'getReturnType')) {
+                $this->fail('The FactoryReturnTypePass should set the class to service "factory".');
+            } else {
+                $this->assertEquals('Please add the class to service "factory" even if it is constructed by a factory since we might need to add method calls based on compile-time checks.', $e->getMessage());
+            }
+        }
```

Comment:
this test will not fail if no RuntimeException is thrown, and there is no assertion in this case